api/python/provisioner/commands/swupgrade.py

```diff
@@ -0,0 +1,147 @@
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU Affero General Public License for more details.
+# You should have received a copy of the GNU Affero General Public License
+# along with this program. If not, see <https://www.gnu.org/licenses/>.
+# For any questions about this software or licensing,
+# please email opensource@seagate.com or cortx-questions@seagate.com.
+#
+import logging
+from typing import Type
+
+from api.python.provisioner import inputs
+from api.python.provisioner.commands import (
+    CommandParserFillerMixin, Check, SWUpdateDecisionMaker,
+    _apply_provisioner_config, _restart_salt_minions, _update_component)
+from api.python.provisioner.config import GroupChecks
+from api.python.provisioner.errors import (
+    SWStackUpdateError, ClusterNotHealthyError,
+    SWUpdateError, SWUpdateFatalError
+)
+from api.python.provisioner.hare import ensure_cluster_is_healthy
+from api.python.provisioner.salt import StatesApplier, local_minion_id
+from api.python.provisioner.salt_master import config_salt_master
+from api.python.provisioner.salt_minion import config_salt_minions
+from api.python.provisioner.vendor.attr import attr
+
+
+logger = logging.getLogger(__name__)
+
+
+COMPONENTS_FOR_UPGRADE = frozenset({'motr',
+                                    's3server',
+                                    'hare',
+                                    'ha.cortx-ha',
+                                    'sspl',
+                                    'uds',
+                                    'csm'})
+
+
+@attr.s(auto_attribs=True)
+class SWUpgrade(CommandParserFillerMixin):
+    input_type: Type[inputs.NoParams] = inputs.NoParams
+
+    @staticmethod
+    def _ensure_upgrade_repos_configuration():
+        logger.info("Ensure update repos are configured")
+        StatesApplier.apply(['components.misc_pkgs.swupgrade.repo'],
+                            local_minion_id())
+
+        # logger.info("Check yum repos are good")
+        # StatesApplier.apply(
+        #     ['components.misc_pkgs.sw_update.repo.sanity_check'],
+        #     local_minion_id()
+        # )
+
+    def run(self, targets):  # noqa: C901 FIXME
+        # logic based on https://jts.seagate.com/browse/EOS-6611?focusedCommentId=1833451&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-1833451  # noqa: E501
+
+        # TODO:
+        #   - create a state instead
+        #   - what about apt and other non-yum pkd managers
+        #   (downgrade is another more generic option but it requires
+        #    exploration of depednecies that are updated)
+        # TODO IMPROVE minions might be stopped here in case of rollback,
+        #      options: set up temp ssh config and rollback yum + minion config
+        #      via ssh as a fallback
+        rollback_ctx = None
+        try:
```

Comment:
[that](https://github.com/Seagate/cortx-prvsnr/blob/7bf919462b747634d54d4f5c6978f94b1ec40f0f/api/python/provisioner/commands/__init__.py#L527-L531) logic of yum rollback in case of any errors is necessary for upgrade as well. Please restore.

Reply:
Done. I restored the previous logic.